modules/uhttp/client/client_filters.go

```diff
@@ -88,13 +88,21 @@ func authenticationFilter(
 	authClient := auth.Instance()
 
 	// Client needs to know what service it is to authenticate
-	authClient.SetAttribute(ctx, auth.ServiceAuth, _serviceName)
+	authctx := authClient.SetAttribute(ctx, auth.ServiceAuth, _serviceName)
 
-	authctx, err := authClient.Authenticate(ctx)
+	authctx, err = authClient.Authenticate(authctx)
 	if err != nil {
 		ctx.Logger().Error(auth.ErrAuthentication, "error", err)
 		return nil, err
 	}
+
+	span := opentracing.SpanFromContext(authctx)
```

Comment:
We should just do this once after all the filters. You can make a spanInjectFilter that just does this and execute tracingFilter, authFilter, spanInjectFilter. So neither tracing nor auth needs to inject it.